packages/ui/src/primitives/tailwind/Input/index.tsx

```diff
@@ -23,141 +23,157 @@ All portions of the code written by the Infinite Reality Engine team are Copyrig
+   * The color of the helper text is determined by the current state.
+   */
+  helperText?: string
+
+  fullWidth?: boolean
 }
 
-const variants = {
-  outlined: ' ',
-  underlined: 'bg-transparent border-0 border-b rounded-none placeholder:text-neutral-200 placeholder:text-[17px]'
+interface LabelProps {
+  labelText: string
+  labelPosition: 'top' | 'left'
+  infoText?: string
 }
 
-const Input = forwardRef<HTMLInputElement, InputProps>(
-  (
-    {
-      className,
-      containerClassName,
-      label,
-      type = 'text',
-      error,
-      description,
-      value,
-      itemType,
-      onChange,
-      disabled,
-      startComponent,
-      endComponent,
-      variant = 'outlined',
-      labelClassname,
-      errorBorder,
-      autoComplete,
-      ...props
-    },
-    ref
-  ) => {
-    // Define a base style for the onboarding variant and conditionally apply error styling
-    const onboardingVariantStyle = errorBorder
-      ? 'bg-transparent border border-rose-600 rounded-none placeholder:text-neutral-200 placeholder:text-[17px]'
-      : 'bg-transparent border border-neutral-500 rounded-none placeholder:text-neutral-200 placeholder:text-[17px]'
-
-    const twClassname = twMerge(
-      'text-base font-normal tracking-tight',
-      'textshadow-sm flex h-9 w-full rounded-lg border border-theme-primary bg-theme-surfaceInput px-3.5 py-5 transition-colors',
-      'file:border-0 file:bg-theme-surfaceInput file:text-sm file:font-medium',
-      'dark:[color-scheme:dark]',
-      'focus-visible:ring-ring placeholder:text-gray-400 focus-visible:outline-none disabled:cursor-not-allowed disabled:opacity-50',
-      variant !== 'outlined' ? '' : 'focus-visible:ring-1',
-      startComponent ? 'ps-10' : undefined,
-      endComponent ? 'pe-10' : undefined,
-      variant === 'onboarding' ? onboardingVariantStyle : variants[variant],
-      className
-    )
-
-    const containerVariants = {
-      outlined: 'gap-2',
-      underlined: '',
-      onboarding: ''
+export type InputProps = InputBaseProps & LabelProps
+
+const InputBase = (
+  {
+    variantSize,
+    startComponent,
+    endComponent,
+    state,
+    helperText,
+    labelText,
+    labelPosition,
+    infoText,
+    required,
+    id,
+    fullWidth,
+    ...props
+  }: InputProps,
+  ref: React.ForwardedRef<HTMLInputElement>
+) => {
+  const containerClass = twMerge(
+    'flex w-full items-center gap-x-2 rounded-md border-[0.5px] border-[#42454D] bg-[#141619] transition-colors duration-300',
+    variantSizes[variantSize],
+    'hover:border-[#9CA0AA] hover:bg-[#191B1F]',
+    'has-[:focus]:border-[#375DAF] has-[:focus]:outline-none',
+    'has-[:disabled]:border-[#42454D] has-[:disabled]:bg-[#191B1F]',
+    state === 'success' && 'border-[#10B981]',
+    state === 'error' && 'border-[#C3324B]'
+  )
+
+  const inputClass = twMerge(
+    'peer order-2 h-full w-full bg-inherit text-[#9CA0AA] outline-none focus:text-[#F5F5F5] disabled:text-[#6B6F78]'
+  )
```

Comment:
twmerge not required

```suggestion
  const inputClass = 'peer order-2 h-full w-full bg-inherit text-[#9CA0AA] outline-none focus:text-[#F5F5F5] disabled:text-[#6B6F78]'
```